netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,605 @@
+from netket.operator._pauli_strings import _count_of_locations
+from netket.hilbert.abstract_hilbert import AbstractHilbert
+import re
+from collections import defaultdict
+
+
+class FermionOperator2nd(DiscreteOperator):
+    """Constructs a fermion operator given the single terms (set of creation/annihilation operators) in second quantization formalism."""
+
+    def __init__(
+        self,
+        hilbert: AbstractHilbert,
+        terms: Union[List[str], List[List[List[int]]]],
+        weights: List[Union[float, complex]] = None,
+        constant: Union[float, complex] = 0.0,
+        dtype: DType = complex,
+    ):
+
+        r"""
+        This class can be initialized in the following form: ``FermionOperator2nd(hilbert, terms, weights ...)``.
+        The terms contain pairs of (idx, dagger), where the idx is the index in the output of the hilbert.all_states()
+        To split up per spin, use the creation and annihilation operators to build the operator.
+
+        Args:
+            hilbert (required): hilbert of the resulting FermionOperator2nd object
+            terms (list(str) or list(list(list(int)))): single term operators (see example below)
+            weights (list(union(float,complex))): corresponding coefficients of the single term operators
+
+        Returns:
+            A FermionOperator2nd object.
+
+        Example:
+            Constructs a new ``FermionOperator2nd`` operator (0.5-0.5j)*(a_0^dagger a_1) + (0.5+0.5j)*(a_2^dagger a_1)  with the construction scheme.
+            >>> import netket.experimental as nkx
+            >>> terms, weights = (((0,1),(1,0)),((2,1),(1,0))), (0.5-0.5j,0.5+0.5j)
+            >>> hi = nkx.hilbert.SpinOrbitalFermions(3)
+            >>> op = nkx.operator.FermionOperator2nd(hi, terms, weights)
+            >>> op
+            FermionOperator2nd(hilbert=SpinOrbitalFermions(n_orbitals=3), n_terms=2)
+            >>> terms = ("0^ 1", "2^ 1")
+            >>> op = nkx.operator.FermionOperator2nd(hi, terms, weights)
+            >>> op
+            FermionOperator2nd(hilbert=SpinOrbitalFermions(n_orbitals=3), n_terms=2)
+            >>> op.hilbert
+            SpinOrbitalFermions(n_orbitals=3)
+            >>> op.hilbert.size
+            3
+        """
+        super().__init__(hilbert)
+        self._dtype = dtype
+        self._orb_idxs = []
+        self._daggers = []
+        self._term_ends = []
+        self._weights = []
+        self._n_terms = 0
+        # we keep the input, in order to be able to add terms later
+        self._orig_terms = []
+        self._orig_weights = []
+        self._constant = constant
+
+        if isinstance(terms, str):
+            terms = (terms,)
+
+        if len(terms) > 0 and isinstance(terms[0], str):
+            terms = list(map(_parse_string, terms))
+
+        if weights is None:
+            weights = [1.0] * len(terms)
+
+        if not len(weights) == len(terms):
+            raise ValueError("length of weights should be equal")
```

Comment:
make this docstrin gbetter. maybe print the length of weights and number of terms